### DISCUSSION TITLE
Resize an Image with ImageMagick Without Losing Quality

### DISCUSSION THREAD

**Discussion by flywire** [ImageMagick/ImageMagick#6245] [Help]
### ImageMagick version

7.1.0-62

### Operating system, version and so on

Win10

### Description

How can I reduce the print size of an image to 25% maintaining aspect and retain all the pixels (ie increase the resolution by 4 times)? Say 300 dpi image to 1200 dpi image 1/4 of print size. I expect this is covered in https://imagemagick.org/Usage/resize/ but I can't see it. 

Using `C:\Program Files\ImageMagick-7.1.0-Q16-HDRI>magick images\logo.png -scale 25% e:\logo2.png` retains the image resolution on 25% print size which is 1/16 of the number of pixels.

Ultimately I want say a scanned image with white background to 1/4 print size at 4 times resolution as a pdf with a transparent background withot specifying resolution.

```
Image:
  Filename: images\logo.png
  Permissions: rw-rw-rw-
  Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: DirectClass
  Geometry: 2135x2048+0+0
  Resolution: 118.11x118.11
  Print size: 18.0764x17.3398
  Units: PixelsPerCentimeter
```

**Comment by snibgo** [comment]
To reduce the "print size" without changing the number of pixels, use "-density". To reduce the print size to 25%, multiply the density by 4. For example, assuming the density is the same in both directions:
```
magick toes.png -density %[fx:%[resolution.x]*4] x.png
```

**Reply by flywire** [reply]
Let's break that down:

https://imagemagick.org/script/command-line-options.php#density

https://www.imagemagick.org/script/escape.php
```
%[key]			percent escape
%[fx:expression]	Calculated Percent Escape Prefixes 
%[resolution.x]		X density (resolution) without units
```

---

> Ultimately I want... a pdf

Density works fine for a png but not for the pdf as shown below:

```
magick images\logo.png -density %[fx:%[resolution.x]*4] test.pdf
magick identify -verbose test.pdf
Image:
  Filename: test.pdf
  Permissions: rw-rw-rw-
  Format: PDF (Portable Document Format)
  Mime type: application/pdf
  Class: DirectClass
  Geometry: 128x123+0+0
  Resolution: 72x72
  Print size: 1.77778x1.70833
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColorAlpha
...
```

**Comment by snibgo** [comment]
>   Units: Undefined

When units are undefined, so is the print size, which is "1.77778x1.70833" but the units could be miles or millimetres.

"`-density %[fx:%[resolution.x]*4]`" still works, multiplying the resolution by 4 and dividing the print size by 4, but when neither have units the result may not be what you want.

**Comment by flywire** [comment]
Can you link me to the GitHub source for https://imagemagick.org/Usage/resize/? I'd like to make a PR to reference density at the start.

---

The discussion conflicts with https://imagemagick.org/script/command-line-options.php#density. Especially for example, "_To resize the image so that it is the same size..._". The statement is illogical and _size_ is used inconsistently. What do you think?

It is important to understand the intended meaning of the term size in images because it is used to describe different things. The size of the canvas, also referred to as print size, can be changed in ImageMagick with the density command which does not change any of the pixels in the image. While some software uses this setting other software ignores it.

The term size is also used for the number of bytes to store the image. Changing this size can vary from less of an image with the same pixels to fewer pixels or less quality on the pixels. Any changes need to consider the intended use of the image as a high level of detail might be needed for an area on a printed page but if that level of detail cannot be seen on a screen the processing required to display it is inefficient.

**Comment by flywire** [comment]
The issue seems similar to the unresolved #3010

**Comment by snibgo** [comment]
I don't know where the source for the https://imagemagick.org/Usage pages are.

I agree the documentation often uses "size" too loosely. It mostly means "the dimensions in pixels" but sometimes means "the dimension in physical units such as inches or centimeters" or "the file size in bytes".

In https://imagemagick.org/script/command-line-options.php#density:

> To resize the image so that it is the same size at a different resolution, use the -resample option.

This might be better expressed as:

> To change the number of pixels in the image so that it has the same physical dimensions (in inches or centimeters), use the -resample option.

This sentence is also wrong:
> To determine the resolution of your display, use a ruler to measure the width of your screen in inches, and divide by the number of horizontal pixels (1024 on a 1024x768 display).

The resolution is, of course, the number of pixels divided by the width in inches.
